src/ae.c

```diff
@@ -503,3 +555,17 @@ void aeSetBeforeSleepProc(aeEventLoop *eventLoop, aeBeforeSleepProc *beforesleep
 void aeSetAfterSleepProc(aeEventLoop *eventLoop, aeAfterSleepProc *aftersleep) {
     eventLoop->aftersleep = aftersleep;
 }
+
+/* This function allows setting a custom poll procedure to be used by the event loop.
+ * The custom poll procedure, if set, will be called instead of the default aeApiPoll */
+void aeSetCustomPollProc(aeEventLoop *eventLoop, aeCustomPollProc *custompoll) {
+    eventLoop->custompoll = custompoll;
+}
+
+void aeSetPollProtect(aeEventLoop *eventLoop, int protect) {
+    if (protect) {
```

Comment:
should this be protected via the mutex as well?

Reply:
The `AE_PROTECT_POLL` flag will be true either at the start or only at the end of the function therefore, we will only perform lock/unlock operations if we are to use `AE_LOCK(eventLoop)` and `AE_UNLOCK(eventLoop)`. 

As we call this function only from the main thread when we are certain we can remove the protection or when we are about to send it to a new thread, meaning we are currently the only thread accessing it there is no need it guarding it.